portfolio/src/main/java/com/google/sps/servlets/models/CommentReader.java

```diff
@@ -0,0 +1,101 @@
+package com.google.sps.servlets.models;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.appengine.api.datastore.TransactionOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public abstract class CommentReader {
```

Comment:
Why is this abstract?  This looks like a util class, although it would be better to instantiate it with an instance of the datastore service (assuming that can be used multiple times in different methods).